jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/RemoteProcessRuntimeDataServiceImpl.java

```diff
@@ -105,7 +105,7 @@ public ProcessInstanceSummary getProcessInstance(String serverTemplateId, Proces
             if(nis.getNodeType().equals("HumanTaskNode")){
                 logs.add(new RuntimeLogSummary(nis.getId(), prettyDateFormatter.format(nis.getDate()), "Task '" + nis.getName() + "' was created", "System"));
                 for(TaskEventInstance te : allTaskEventsByProcessInstanceId){
-                    if(te.getWorkItemId() != null && nis.getId() == te.getWorkItemId() &&
+                    if(te.getWorkItemId() != null && te.getWorkItemId().equals(nis.getId()) &&
```

Comment:
This and the following methods compare two reference values using the == or != operator, where the correct way to compare instances of this type is generally with the equals() method.